src/stcal/ramp_fitting/ramp_fit.py

```diff
@@ -212,9 +229,47 @@ def ramp_fit_data(ramp_data, buffsize, save_opt, readnoise_2d, gain_2d,
         nframes = ramp_data.nframes
         readnoise_2d *= gain_2d / np.sqrt(2. * nframes)
 
+        # Suppress one group ramps, if desired.
+        if ramp_data.suppress_one_group_ramps:
+            suppress_one_group_ramps(ramp_data)
+
         # Compute ramp fitting using ordinary least squares.
         image_info, integ_info, opt_info = ols_fit.ols_ramp_fit_multi(
             ramp_data, buffsize, save_opt, readnoise_2d, gain_2d, weighting, max_cores)
         gls_opt_info = None
 
     return image_info, integ_info, opt_info, gls_opt_info
+
+
+def suppress_one_group_ramps(ramp_data):
+    """
+    Finds one group ramps in each integration and suppresses them, i.e. turns
+    them into zero group ramps.
+
+    Parameter
+    ---------
+    model : data model
+        input data model, assumed to be of type RampModel
+    """
+    dq = ramp_data.groupdq
+    nints, ngroups, nrows, ncols = dq.shape
+
+    for k in range(nints):
+        intdq = dq[k, :, :, :]
+        good_groups = np.zeros(intdq.shape, dtype=int)
+
+        # Compute the number of good groups in each ramp
+        good_groups[intdq == 0] = 1
+        ngood_groups = good_groups.sum(axis=0)
+
+        # For each pixel that has only one good group, mark those groups as
+        # DO_NOT_USE, making the ramp effectively zero good groups.
```

Comment:
Is this DO_NOT_USE setting only being applied to some internal data array, for use solely within the slope calculations? We don't want those flags to get set in the original input data to the step (i.e. we don't want to make them permanent).